core/src/main/scala/kafka/admin/AdminClient.scala

```diff
@@ -137,6 +151,71 @@ class AdminClient(val time: Time,
       broker -> Try[NodeApiVersions](new NodeApiVersions(getApiVersions(broker).asJava))
     }.toMap
 
+
+  /*
+   * Remove all the messages whose offset is smaller than the given offset of the corresponding partition
+   *
+   * PurgeDataResult contains either lowWatermark of the partition or exception. We list the possible exception
+   * and their interpretations below:
+   *
+   * - DisconnectException if leader node of the partition is not available. Need retry by user.
+   * - TopicAuthorizationException if the topic doesn't exist and the user doesn't have the authority to create the topic
+   * - TimeoutException if response is not available within the timeout specified by either Future's timeout or AdminClient's request timeout
+   * - UnknownTopicOrPartitionException if the partition doesn't exist or if the user doesn't have the authority to describe the topic
+   * - NotLeaderForPartitionException if broker is not leader of the partition. Need retry by user.
+   * - OffsetOutOfRangeException if the offset is larger than high watermark of this partition
+   *
+   */
+
+  def purgeDataBefore(purgeOffsets: Map[TopicPartition, Long]): Future[Map[TopicPartition, PurgeDataResult]] = {
```

Comment:
@ijuma Have we explicitly stated in the code or documentation that this class is internal and should not be used by users? Becket is not aware of this as well. It seems that user can already construct Scala AdminClient directly and that is how I expect this API to be used.